PortalFetch/crawler_sel.py

```diff
@@ -0,0 +1,332 @@
+                "Section Search".lower() == ipt.get_attribute("value").lower():
+            submit = ipt
+            break
+    if not submit:
+        logging.info("No submit button found! ! ")
+        sys.exit(-1)  # Program exit
+    #Click the submit button
+    submit.click()
+    # Wait for a while, wait for the page to open and make sure it loads
+    time.sleep(3)
+
+
+# Save the results of the course
+def saveResult(driver):
+    """
+    Save result to a html.
+
+    Parameters: webdriver
+    Returns: html
+    Return type: String
+    """
+    waitUtilPageLoaded(driver)
+    # Scroll down a page to show the table (no practical meaning, just look at the effect)
+    actions = ActionChains(driver)
+    actions.send_keys(Keys.PAGE_DOWN).perform()
+    # After loading is complete, the entire form page
+    html = driver.page_source
+    time.sleep(5)  # Wait for a while, let people take a look, in fact, it does not make sense to the programmer
+    return html
+
+
+# Wait until the page is loaded! This only applies to child pages, not to the main page, because there is no banner_copyright information in the main page
+def waitUtilPageLoaded(driver):
+    """
+    Wait until page loaded.
+
+    Parameters: webdriver
+    Returns: None
+    """
+    while True:
+        try:
+            driver.find_element_by_class_name("banner_copyright")
+            break
+        except:
+            logging.info("retry after 1 second...")
+            time.sleep(1)
+            pass
+
+
+# input: text is html of the courses, ulist is an empty list []
+# output : ulist will include all the courses list in text
+def get_contents(ulist, text):
+    """
+    Get result contents from html.
+
+    Parameters: List, String
+    Returns: None
+    """
+    soup = BeautifulSoup(text, 'lxml')
+    trs = soup.find_all('tr')
+    for tr in trs:
+        ui = []
+        for td in tr:
+            ui.append(td.string)
+        ulist.append(ui)
+
+
+# input: urlist is the courses list we get from get_contents
+# output : save courses in a file "DeAnza2020spring.csv"
+def save_contents(filename, firstline, urlist):
+    """
+    Save course result to a .csv file.
+
+    Parameters: String, String, List
+    Returns: None
+    """
+    with open(filename, 'w', encoding='utf-8', newline='') as f:
+        writer = csv.writer(f)
+        writer.writerow([firstline])
+
+        for i in range(8,len(urlist)-1):
+            if len(urlist[i]) >=35:
+                writer.writerow([urlist[i][1], urlist[i][3], urlist[i][5],urlist[i][7], urlist[i][9], urlist[i][11],
+                            urlist[i][13], urlist[i][15], urlist[i][17],urlist[i][19], urlist[i][21], urlist[i][23],
+                            urlist[i][25], urlist[i][27], urlist[i][29],urlist[i][31], urlist[i][33], urlist[i][35]]
+                            )
+            else:
+                writer.writerow(urlist[i])
+    logging.info("Finished!")
```

Comment:
What is finished? Just in case other team members might want to know by looking through the log file when they are doing trouble shootings.